### PR TITLE
Increasing test coverage

### PR DESCRIPTION
**Pull request by luceos**
**Confirmed**

- [ ] ~~Frontend changes: tested on a local Flarum installation.~~
- [x] Backend changes: tests are green (run `php vendor/bin/phpunit`).

### CODE REVIEW

**Comment by franzliedke** [comment]
@luceos Still WIP or ready to go?

**Comment by luceos** [comment]
Let me add the error reporting level as a todo here before we merge @franzliedke

**Comment by franzliedke** [comment]
Okay, great: :+1:

**Comment by luceos** [comment]
@franzliedke added: https://travis-ci.org/flarum/core/jobs/470697812#L660

```
32767 | E_ALL (integer)
```

**Comment by luceos** [comment]
Ready to be merged if green ;)

File: tests/Api/Controller/ListGroupsControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     * @test
+     */
+    public function shows_index_for_guest()
+    {
+        $response = $this->callWith();
+
+        $this->assertEquals(200, $response->getStatusCode());
+        $data = json_decode($response->getBody()->getContents(), true);
+
+        $this->assertEquals(Group::count(), count($data['data']));
```
Comment (franzliedke) [inline tests/Api/Controller/ListGroupsControllerTest.php]: Shouldn't this at least set up a group or something? I'd rather have the expected value be a fixed value, e.g. three groups.

**Comment by franzliedke** [comment]
Merging for now - I want to work on restructuring the tests a bit.

File: tests/Api/Controller/ListGroupsControllerTest.php (hunk 1 above)
Comment (luceos) [reply tests/Api/Controller/ListGroupsControllerTest.php]: Yes we should go deeper into confirming actual values deeper down. Sorry I didn't have the time to patch this up yet before you merged.

Comment (franzliedke) [reply tests/Api/Controller/ListGroupsControllerTest.php]: No worries. Want to improve the infrastructure first. :)

**Pull request merged**